flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,273 @@
+
+/** Tests Knn and KnnModel. */
+public class KnnTest {
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+    private Table trainData;
+    private static final String LABEL_COL = "test_label";
+    private static final String PRED_COL = "test_prediction";
+    private static final String VEC_COL = "test_features";
+    private static final List<Row> trainArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(1, Vectors.dense(2.0, 3.0)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(2, Vectors.dense(200.1, 300.1)),
+                            Row.of(2, Vectors.dense(200.2, 300.2)),
+                            Row.of(2, Vectors.dense(200.3, 300.3)),
+                            Row.of(2, Vectors.dense(200.4, 300.4)),
+                            Row.of(2, Vectors.dense(200.4, 300.4)),
+                            Row.of(2, Vectors.dense(200.6, 300.6)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.3, 3.2)),
+                            Row.of(1, Vectors.dense(2.3, 3.2)),
+                            Row.of(3, Vectors.dense(2.8, 3.2)),
+                            Row.of(4, Vectors.dense(300., 3.2)),
+                            Row.of(1, Vectors.dense(2.2, 3.2)),
+                            Row.of(5, Vectors.dense(2.4, 3.2)),
+                            Row.of(5, Vectors.dense(2.5, 3.2)),
+                            Row.of(5, Vectors.dense(2.5, 3.2)),
+                            Row.of(1, Vectors.dense(2.1, 3.1))));
+
+    private static final List<Row> testArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(5, Vectors.dense(4.0, 4.1)), Row.of(2, Vectors.dense(300, 42))));
+    private Table testData;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+
+        Schema schema =
+                Schema.newBuilder()
+                        .column("f0", DataTypes.INT())
+                        .column("f1", DataTypes.of(DenseVector.class))
+                        .build();
+
+        DataStream<Row> dataStream = env.fromCollection(trainArray);
+        trainData = tEnv.fromDataStream(dataStream, schema).as(LABEL_COL + "," + VEC_COL);
```

Comment:
While this approach works, it seems less efficient to require the underlying implementation to parse the schema from the string.

Could we use `as(LABEL_COL, VEC_COL)` here?

Reply:
OK